explorer/src/components/account/AnchorAccountCard.tsx

```diff
@@ -0,0 +1,176 @@
+    return <ErrorCard text={"Failed to find anchor account discriminator"} />;
+  }
+
+  let foundAccountLayoutName: string | undefined;
+  let decodedAnchorAccountData: Object | undefined;
+  if (program) {
+    // Iterate all the structs, see if any of the name-hashes match
+    Object.keys(program.account).forEach((accountType) => {
+      const layoutName = capitalizeFirstLetter(accountType);
+      const discriminatorToCheck =
+        BorshAccountsCoder.accountDiscriminator(layoutName);
+
+      if (discriminatorToCheck.equals(discriminator)) {
+        foundAccountLayoutName = layoutName;
+        const accountDecoder = program.account[accountType];
+        decodedAnchorAccountData = accountDecoder.coder.accounts.decode(
+          layoutName,
+          accountBuffer
+        );
+      }
+    });
+  }
+
+  if (!foundAccountLayoutName || !decodedAnchorAccountData) {
+    return (
+      <ErrorCard
+        text={
+          "Failed to find matching anchor account type for account discriminator"
+        }
+      />
+    );
+  }
+
+  return (
+    <>
+      <div className="card">
+        <div className="card-header">
+          <div className="row align-items-center">
+            <div className="col">
+              <h3 className="card-header-title">{foundAccountLayoutName}</h3>
+            </div>
+          </div>
+        </div>
+
+        <div className="table-responsive mb-0">
+          <table className="table table-sm table-nowrap card-table">
+            <thead>
+              <tr>
+                <th className="w-1 text-muted">Key</th>
+                <th className="text-muted">Value</th>
+              </tr>
+            </thead>
+            <tbody className="list">
+              {decodedAnchorAccountData &&
+                Object.keys(decodedAnchorAccountData).map((key) => (
+                  <AccountRow
+                    key={key}
+                    valueName={key}
+                    // @ts-ignore
+                    value={decodedAnchorAccountData[key]}
+                  />
+                ))}
+            </tbody>
+          </table>
+        </div>
+        <div className="card-footer">
+          <div className="text-muted text-center">
+            {decodedAnchorAccountData &&
+            Object.keys(decodedAnchorAccountData).length > 0
+              ? `Decoded ${Object.keys(decodedAnchorAccountData).length} Items`
+              : "No decoded data"}
+          </div>
+        </div>
+      </div>
+    </>
+  );
+}
+
+function AccountRow({ valueName, value }: { valueName: string; value: any }) {
+  let displayValue: JSX.Element;
+  if (value && value.constructor && value.constructor.name === "PublicKey") {
+    displayValue = <Address pubkey={value} link />;
+  } else if (value && value.constructor && value.constructor.name === "BN") {
+    displayValue = <>{value.toString()}</>;
+  } else if (value && typeof value !== "object") {
+    displayValue = <>{String(value)}</>;
+  } else if (value) {
```

Comment:
```suggestion
  let displayValue: JSX.Element;
  if (value instanceof PublicKey) {
    displayValue = <Address pubkey={value} link />;
  } else if (value instanceof BN) {
    displayValue = <>{value.toString()}</>;
  } else if (value instanceof Object) {
    displayValue = <>{String(value)}</>;
  } else if (value) {
```

I think this can also be changed to `instanceof`?

Reply:
Agreed- converted checks in new commit & renamed `decodeObject` to `stringifyPubkeyAndBigNum`